### PR TITLE
[Solidity] Bug fix and New Feature Support

### PR DESCRIPTION
**Pull request by ChenfengWei0**
This PR includes
- Bug fix for constructor (symbol name)
- Support function returns (previously only allows return `uint8`)
- Support Fixed and Dynamic Array, e.g.
```solidity
// dyn array with uninitialized size
int[] x;
x = new int[](10);
```
- Code refactoring
- Add more documentation

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

File: src/solidity-frontend/solidity_grammar.cpp [hunk 1]
```diff
@@ -269,11 +291,11 @@ ElementaryTypeNameT get_elementary_type_name_t(const nlohmann::json &type_name)
...
-     * For Literal, their typeString is like "int_const 100".
-     * There is no additional type info (bitsize, signed/unsigned),
-     * This means it will require additional type info from the parent
-     * expr to create an internal type.
-     */
+       * For Literal, their typeString is like "int_const 100".
+       * There is no additional type info (bitsize, signed/unsigned),
+       * This means it will require additional type info from the parent
+       * expr to create an internal type.
+       */
```
Comment (fbrausse) [inline src/solidity-frontend/solidity_grammar.cpp]: This formatting change looks wrong. If performed by clang-format, some earlier definition caused it to believe indentation would be one level higher.

Comment (ChenfengWei0) [reply src/solidity-frontend/solidity_grammar.cpp]: Will fix. Thanks for pointing this out.

**Comment by lucasccordeiro** [comment]
@ChenfengWei0: could you please fix the CI?

**Pull request merged**